docs/tutorials/sparse/train_gluon.md

```diff
@@ -0,0 +1,469 @@
+
+We will use [`timeit`](https://docs.python.org/2/library/timeit.html) to check the performance of these two variants, and analyse some [MXNet Profiler](https://mxnet.incubator.apache.org/versions/master/tutorials/python/profiler.html) traces that have been created from these benchmarks. Additionally, we will inspect the memory usage of the weights (and gradients) using the `print_memory_allocation` function defined below:
+
+
+```python
+def print_memory_allocation(net, block_idxs):
+    blocks = [net[block_idx] for block_idx in block_idxs]
+    weight_nbytes = [get_nbytes(b.weight.data()) for b in blocks]
+    weight_nbytes_pct = [b/sum(weight_nbytes) for b in weight_nbytes]
+    weight_grad_nbytes = [get_nbytes(b.weight.grad()) for b in blocks]
+    weight_grad_nbytes_pct = [b/sum(weight_grad_nbytes) for b in weight_grad_nbytes]
+    print("Memory Allocation for Weight:")
+    for i in range(len(block_idxs)):
+        print('{:7.3f} MBs ({:7.3f}%) for {:<40}'.format(weight_nbytes[i]/1000000,
+                                                              weight_nbytes_pct[i]*100,
+                                                              blocks[i].name))
+    print("Memory Allocation for Weight Gradient:")
+    for i in range(len(block_idxs)):
+        print('{:7.3f} MBs ({:7.3f}%) for {:<40}'.format(weight_grad_nbytes[i]/1000000,
+                                                              weight_grad_nbytes_pct[i]*100,
+                                                              blocks[i].name))
+```
+
+### Benchmark: `FullyConnected`
+
+
+```python
+net = mx.gluon.nn.Sequential()
+net.add(
+    mx.gluon.nn.Dense(in_units=num_features, units=128),
+    mx.gluon.nn.Activation('sigmoid'),
+    mx.gluon.nn.Dense(in_units=128, units=8),
+    mx.gluon.nn.Activation('sigmoid'),
+    mx.gluon.nn.Dense(in_units=8, units=1),
+    mx.gluon.nn.Activation('sigmoid'),
+)
+net.initialize(ctx=ctx)
+trainer = mx.gluon.Trainer(net.collect_params(), 'sgd')
+loss_fn = mx.gluon.loss.SigmoidBinaryCrossEntropyLoss()
+```
+
+
+```python
+%%timeit
+data_iter.reset()
+for batch in data_iter:
+    data = batch.data[0]
+    data = data.as_in_context(ctx)
+    label = batch.label[0].as_in_context(ctx)
+    with mx.autograd.record():
+        pred = net(data)
+        loss = loss_fn(pred, label)
+    loss.backward()
+    trainer.step(data.shape[0])
+    loss.wait_to_read()
+```
+
+    532 ms ± 3.47 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
+
+
+![fully_connected](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/ndarray/sparse/fully_connected.png)
```

Comment:
You can put human text in there instead of underscores if you want. Might be better for screen readers...

Reply:
Updated.